### ISSUE TITLE
[bug] Use a PAT for scorecard action

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/slsa-github-generator#1259]
The current GitHub Actions tokens don't support lookups for branch protection rules. This causes errors in the scorecard results.

```
internal error: error during branchesHandler.setup: internal error: githubv4.Query: Resource not accessible by integration
```

Instead we can authenticate with with a PAT
https://github.com/ossf/scorecard-action#authentication-with-pat

**Comment by ianlewis** [comment]
Related to #547

**Comment by ianlewis** [comment]
@laurentsimon says using a PAT isn't recommended because a PAT needs write access to the repo.

Maybe fine-grained PAT could work? I don't see how to restrict a fine-grained PAT to a org repo though.

**Comment by laurentsimon** [comment]
fine-grained PATs currently don't support graphQL, so it won't work - scorecard uses graphQL extensively. Fine-grained PATs, IIUC, currently always expire, which means it's hard to uses them in a workflow - this restriction may change once the feature is GA, though.
I'm not against using a PAT: pull request won't have access to them anyway, and only maintainer of the repo can access it.

**Comment by ianlewis** [comment]
Yeah, lots of issues with the fine grained PATs. I hope they resolve them eventually. I think I'll just keep this open for now since the reasoning for not using a PAT is sound, and we are able to achieve a reasonably high score without it.
